### ISSUE TITLE
clang-tidy config file support

### ISSUE THREAD

**Issue by danielbisar** [microsoft/vscode-cpptools#9981]
### Feature Request

clang-tidy has the option to specify a config file. This comes in very handy when working with multiple repositories.

The option is command line option `--config-file=<string>`.

This option currently cannot be used via the vscode extension.

**Comment by michelleangela** [comment]
@danielbisar 
The extension has the setting `C_Cpp.codeAnalysis.clangTidy.args` for passing command ling arguments to the clang-tidy executable. Could you try adding the option `--config-file=<string>` to this setting and see if that works?

**Comment by danielbisar** [comment]
@michelleangela clang-tidy config will not work.

```
--config=<string>              - 
                                   Specifies a configuration in YAML/JSON format:
                                     -config="{Checks: '*',
                                               CheckOptions: {x: y}}"
                                   When the value is empty, clang-tidy will
                                   attempt to find a file named .clang-tidy for
                                   each source file in its parent directories.
```
So in contrast to clang-format you cannot specify a filename to config file with this. (clang-format would allow file:PATH/TO/CONFIG). 

Adding --config-file=<string> to C_Cpp.codeAnalysis.clangTidy.args kind of works, but it doesn't allow environment variables to be replaced. Neither the vscode syntax `${env:VARIABLE}` nor bash syntax $VARIABLE works.

**Comment by michelleangela** [comment]
I don't think the extension has logic to parse or resolve environment variables passed to clang-tidy args. @sean-mcmanus could you confirm this? Is there a current workaround in case this is not supported in the near future?

**Comment by sean-mcmanus** [comment]
I've added code to resolve the clangTidy.args variables: https://github.com/microsoft/vscode-cpptools/pull/10000 -- is that sufficient or do you want to keep this issue open to track adding a new setting?

**Comment by danielbisar** [comment]
If that allows me to use variables like in other places in vscode (https://code.visualstudio.com/docs/editor/variables-reference) this is sufficient for me.

**Comment by sean-mcmanus** [comment]
@danielbisar It'll work for all the variable expansion mechanisms we currently support, i.e. anything that currently works with C_Cpp.codeAnalysis.clangTidy.path or "defines" or "includePath" in c_cpp_properties.json, etc. We have some open issues tracking some VS Code variables that aren't supported yet.

**Comment by sean-mcmanus** [comment]
@danielbisar We support `$env` variables and `${workspaceFolder}` and `${workspaceFolderBasename}`, but others in https://code.visualstudio.com/docs/editor/variables-reference may need to explicitly added. Let us know if there are any variable expansions you'd want added -- we might need to add a separate issue to track those though.

**Comment by Zingam** [comment]
> I've added code to resolve the clangTidy.args variables: #10000 -- is that sufficient or do you want to keep this issue open to track adding a new setting?

I'd prefer to have a new setting.

**Comment by sean-mcmanus** [comment]
@Zingam Yeah, equivalent to `--config-file=`. I've created a new issue to track that at https://github.com/microsoft/vscode-cpptools/issues/10006 .

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3 , i.e. the ${workspaceFolder} variable is resolved. https://github.com/microsoft/vscode-cpptools/issues/10006 is tracking adding a new setting.
